opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -1516,7 +1516,7 @@ updateAndCommunicateGroupData(const int reportStepIdx,
                     break;
                 }
                 default:
-                    assert(false); //programming error
+                    throw std::logic_error("MULTI-phase injection is not supported, but was requested for well " + well->name());
```

Comment:
This is fine, of course, but for what it's worth we already have [libfmt](https://github.com/atgeirr/opm-simulators/blob/df45c8b5d92b27f54a19b4f9afcb9d06bac5e2ad/opm/simulators/wells/BlackoilWellModelGeneric.cpp#L87) in this file so there's nothing to lose by using that.